From 1170801b41df85907463fdd6fd750e4f66398059 Mon Sep 17 00:00:00 2001
From: Zhang Yanmin <yanmin.zhang@intel.com>
Date: Thu, 25 Jun 2015 11:28:08 +0800
Subject: [PATCH] EM/Charger: change the return value of
 bq24192_enable_charging

bq24192_enable_charging calls bq24192_read_reg, which might
return a positive value. At s3 suspending, bq24192_suspend
calls bq24192_enable_charging. If bq24192_enable_charging
returns a positive value such like 68, s3 suspend would abort.

The patch fixes it by reset ret to 0 if ret > 0.

Change-Id: I64714a5f5d5822c55f6d1fc7460b63d54c28b3eb
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-39520
Signed-off-by: Zhang Yanmin <yanmin.zhang@intel.com>
---
 drivers/power/bq24192_charger.c | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/drivers/power/bq24192_charger.c b/drivers/power/bq24192_charger.c
index a91d650f85a8..81302fac9d69 100644
--- a/drivers/power/bq24192_charger.c
+++ b/drivers/power/bq24192_charger.c
@@ -1294,12 +1294,14 @@ static inline int bq24192_enable_charging(
 		if (ret < 0) {
 			dev_warn(&chip->client->dev,
 				"read reg failed %s\n", __func__);
-		}
-
-		/* If no charger connected, cancel the workers */
-		if (!(ret & SYSTEM_STAT_VBUS_OTG)) {
-			dev_info(&chip->client->dev, "NO charger connected\n");
-			cancel_delayed_work_sync(&chip->chrg_task_wrkr);
+		} else {
+			/* If no charger connected, cancel the workers */
+			if (!(ret & SYSTEM_STAT_VBUS_OTG)) {
+				dev_info(&chip->client->dev,
+				"NO charger connected\n");
+				cancel_delayed_work_sync(&chip->chrg_task_wrkr);
+			}
+			ret = 0;
 		}
 	}
 
-- 
1.9.1

